server/ast/expr.go

```diff
@@ -105,7 +106,7 @@ func nodeExpr(ctx *Context, node tree.Expr) (vitess.Expr, error) {
 		return nil, fmt.Errorf("ANNOTATE_TYPE is not yet supported")
 	case *tree.Array:
 		unresolvedChildren := make([]vitess.Expr, len(node.Exprs))
-		var coercedType pgtypes.DoltgresType
+		var coercedType sql.Type
```

Comment:
Would this not be `*pgtypes.DoltgresType`?

Reply:
I could also add this check in `*DoltgresType` type assertion.
```
if dt == nil {
  continue
}
```

Reply:
I see, I think the empty check makes more sense from a logical perspective, since relying on a specific kind of nil could be tricky to debug.